### DISCUSSION TITLE
Disabling magic link but allowing email signup using password

### DISCUSSION THREAD

**Discussion by Murkrage** [supabase/supabase#7491] [Questions] [answered]
Hey! 

Today I asked the following question in the Discord #Help channel:

> Is there a way to disable magic link sign ups / sign ins?

The answer was yes, by disabling email signup. However, that completely disables any form of email related sign up and not just the magic link. 

So I'm wondering if there's a way to properly disable the magic link usage while enabling the email/password option. The reason I want to do this is because I'm building a PWA which, unfortunately, doesn't properly handle magic links currently.

One thing I've also noticed is that using the password reset method and subsequently clicking the link that is sent to you, it will log you in. Is this the expected behaviour? I would expect the user to first change their password and then use that password to log into the application.

**Comment by GaryAustin1** [comment, marked as answer]
If your client code requires a password, and you use the password option with the email in your signin then I don't believe a "magiclink" would be sent.   So a user would have to bypass your client logic and issue a auth signin call without the password to get a magiclink that may or may not work, but is not a security risk.

You can take the confirmationURL out of the templates also.

I have a PWA in the works and can't handle redirects, so I plan to use the new token option in signup for email confirmation.  I go to a confirmation page after signup that says to enter the code in your email.  They get the code in the email and copy it over, then I confirm it with verifyOTP similar to the way phone/otp worked.  This link has more info on this (there may be better info, but I found this quickly) https://github.com/supabase/gotrue/issues/264

For signin I require both an email and password.

**Reply by Murkrage** [reply]
@GaryAustin1 Thanks for the info! I think the following comment, only a day old, is exactly what I need: https://github.com/supabase/gotrue/issues/264#issuecomment-1169487153
